src/octoprint/util/comm.py

```diff
@@ -2616,23 +2603,29 @@ def _detect_port(self):
 			return potentials[0]
 
 		elif len(potentials) > 1:
-			programmer = stk500v2.Stk500v2()
 
 			for p in potentials:
-				serial_obj = None
+				self._log("Trying {}".format(p))
 
+				serial_obj = None
+				try:
+				        serial_obj = serial.Serial(str(p), baudrate, timeout=1, writeTimeout=10000)
+				except Exception:
+					self._logger.info("Could not connect to {} ({}).".format(p, baudrate))
+					continue
 				try:
-					self._log("Trying {}".format(p))
-					programmer.connect(p)
-					serial_obj = programmer.leaveISP()
+					if _send_message(serial_obj, [0x30]) != b'\x00\n':
```

Comment:
We should try this up to 5 times. This should fix a reconnect after a  disconnection.

Reply:
Fixed in 08dd797